free-tunes/LogIn/LoginViewController.swift

```diff
@@ -55,19 +46,18 @@ class LoginViewController: UIViewController {
         passwordInput.applyDefaultStyle(withName: "Password")
         passwordInput.delegate = self
     }
+}
 
-// MARK: - Input Validation
-    func checkTextFieldData() -> Bool {
-        if usernameInput.isEmpty && passwordInput.isEmpty {
-            emptyTextFields()
-            return false
-        }
-        
-        return true
-    }
-
-    func emptyTextFields() {
+extension LoginViewController: LogInViewModelDelegate {
+    
+    func notify() {
         usernameInput.emptyFieldError()
```

Comment:
Rather use your delegate functions to determine what happens 😄 for example do you checks in your `ViewModel` and then if all the checks pass call `notify()` telling it to perform a `segue` or call `show(error: String)` to let the user know what went wrong.

Reply:
Also maybe think of a more descriptive name for `notify()` it is a little bit vague.

Reply:
I have updated the code in a way I think achieves the above outlined suggestions😄👍.